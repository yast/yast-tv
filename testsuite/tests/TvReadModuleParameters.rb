# encoding: utf-8

module Yast
  class TvReadModuleParametersClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # Read() return nil
      @READ_nil = { "modules" => { "options" => nil } }

      # Read() returns empty map
      @READ_empty = { "modules" => { "options" => {} } }

      # Read() returns options for one card
      @READ_one = {
        "modules" => {
          "options" => { "bttv" => { "card" => "2", "tuner" => "5" } }
        }
      }

      # Read() returns options for three cards
      @READ_three = {
        "modules" => {
          "options" => {
            "bttv" => { "card" => ",,3", "tuner" => "a,b,c", "blah" => ",y," }
          }
        }
      }

      TEST(lambda { Tv.ReadModuleParameters(path(".modules"), "bttv") }, [
        @READ_nil,
        {},
        {}
      ], false)

      TEST(lambda { Tv.ReadModuleParameters(path(".modules"), "bttv") }, [
        @READ_empty,
        {},
        {}
      ], false)

      TEST(lambda { Tv.ReadModuleParameters(path(".modules"), "bttv") }, [
        @READ_one,
        {},
        {}
      ], false)

      #    TEST(``(Tv::ReadModuleParameters ("card")), [ READ_three , $[], $[] ], false);
      TEST(lambda { Tv.ReadModuleParameters(path(".modules"), "bttv") }, [
        @READ_three,
        {},
        {}
      ], false)

      nil
    end
  end
end

Yast::TvReadModuleParametersClient.new.main
