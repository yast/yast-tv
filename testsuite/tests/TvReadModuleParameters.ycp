{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    // Read() return nil
    map READ_nil =
	$[ "modules" :
	    $[ "options" : nil ]
	];

    // Read() returns empty map
    map READ_empty =
	$[ "modules" :
	    $[ "options" : $[] ]
	];

    // Read() returns options for one card
    map READ_one =
	$[ "modules" :
	    $[ "options" : $[
            "bttv": $[ "card" : "2",
			      "tuner" : "5" ]
                  ]
	    ]
	];

    // Read() returns options for three cards
    map READ_three =
	$[ "modules" :
	    $[ "options" :
            $[ "bttv": $[ "card" :  ",,3",
			      "tuner" : "a,b,c",
			      "blah":   ",y," ]
            ]
	    ]
	];

    TEST(``(Tv::ReadModuleParameters ("bttv")), [ READ_nil, $[], $[] ], false);

    TEST(``(Tv::ReadModuleParameters ("bttv")), [ READ_empty , $[], $[] ], false);

    TEST(``(Tv::ReadModuleParameters ("bttv")), [ READ_one , $[], $[] ], false);

//    TEST(``(Tv::ReadModuleParameters ("card")), [ READ_three , $[], $[] ], false);
    TEST(``(Tv::ReadModuleParameters ("bttv")), [ READ_three , $[], $[] ], false);
}
