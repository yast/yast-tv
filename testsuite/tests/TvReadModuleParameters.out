Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.UseSSH nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.InstMode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dir	.modules.options: []
Return	$[]
Dir	.modules.options: []
Return	$[]
Dir	.modules.options: ["bttv"]
Read	.modules.options."bttv" $["card":"2", "tuner":"5"]
Return	$["card":$[0:"2"], "tuner":$[0:"5"]]
Dir	.modules.options: ["bttv"]
Read	.modules.options."bttv" $["blah":",y,", "card":",,3", "tuner":"a,b,c"]
Return	$["blah":$[0:"", 1:"y", 2:""], "card":$[0:"", 1:"", 2:"3"], "tuner":$[0:"a", 1:"b", 2:"c"]]
